django_rehive_extras/fields.py

```diff
@@ -67,6 +67,18 @@ def generate_content(self, instance, content):
             return self.clear_metadata(instance, content)
         return content
 
+    @classmethod
+    def _clear_metadata(cls, instance, file):
+        try:
+            pyexiv_image = ImageData(file.read())
+            pyexiv_image.clear_exif()
+            file.seek(0)
+            file.truncate()
+            file.write(pyexiv_image.get_bytes())
+            return file
+        except Exception as e:
```

Comment:
Hi @almabud2 I do not like that this has a general `Exception`. You need to find out what exception gets thrown if a non image file is handled instead of just having a general exception that will catch anything.

Reply:
HI, @OdinsPlasmaRifle if I added specific `Exception` then I have to `catch` other `Exception` too. Otherwise, this will no file will be saved or some exceptions will be raised that remain unhandled.

- So, if this is not an `image` file then `I have to return the original file`. This will happen for the `specified Exception`. 
- But If I catch other exceptions then I can `raise Exception` or `return` the `original` file. Looks like this ->
```
        try:
            pyexiv_image = ImageData(file.read())
            pyexiv_image.clear_exif()
            file.seek(0)
            file.truncate()
            file.write(pyexiv_image.get_bytes())
            return file
        except NoneImageFileException as e:
            return file
        except OtherExcpetion as e:
            return file
```
But then what is the purpose of `specified Exception` if we return the `original` file for the `both` exception?

Reply:
> Otherwise, no files will be saved or some exceptions will be raised that remain unhandled.
This is what we want to happen though? You should have explicit `try...except` whenever possible, otherwise you will silently be catching errors that you do not know about.

If you catch every exception with a broad `Exception` then you will catch all errors, which means we will never get sentry notifications about errors we are not aware of. In what circumstance will you get exceptions of a type you do not expect? and if you know the types then why not include them in the exception handler?

I don't want a general `Exception` catch, I want those errors to be raise and be unhandled  because we need to know about those errors and they should trigger a 500 error.

What I want is:
```
        try:
            pyexiv_image = ImageData(file.read())
            pyexiv_image.clear_exif()
            file.seek(0)
            file.truncate()
            file.write(pyexiv_image.get_bytes())
            return file
        # If the file is not an image then simply return the file.
        except NoneImageFileException as e:
            return file
            
       # ALL other errors are unhandled as we want to 500 errors on these or add them as additional explicit handlers.
```

>  Otherwise, **no files will be saved** or some exceptions will be raised that remain unhandled.

I don;t understand this, why would "no other files be saved".... the only files that will not be saved will be ones that trigger an unexpected error, or one we have not accounted for. Which is exactly what we want.

Reply:
It is the same concept as something like this:

```python
try:
    thing = Thing.objects.get(id=1)
except Thing.DoesNotExist:
    raise NotFound()
```

We only catch `DoesNotExist` because that is an error we expect to see there.  If we caught `Exception` then we would potentially catch all sorts of other errors like: database connection error, class not imported errors, class method errors, unexpected field format errors. But we want to know about these other errors so we only catch the DoesNotExist instead.